### PR TITLE
Pre-processing of CWL workflows and tools for conformance tests on PAPI

### PR DESCRIPTION
**Pull request by Horneth**
I removed the default gcs prefix and default docker conf thinking that they only existed for conformance tests but maybe we want to keep them for other purposes ?

### CODE REVIEW

File: centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  private def processYaml(value: String)(f: Json => Json) = process(value, f, yaml.Printer.spaces2.pretty)
+
+  // Process and print back as JSON
+  private def processJson(value: String)(f: Json => Json) = process(value, f, io.circe.Printer.spaces2.pretty)
+
+  // Prefix the string at "key" with the gcs prefix
+  private def prefixLocationWithGcs(key: String): Json => Json = root.selectDynamic(key).string.modify(gcsPrefix + _)
+
+  // Prefix "location" and "path"
+  private val prefix = prefixLocationWithGcs("location").compose(prefixLocationWithGcs("path"))
```
Comment (cjllanwarne) [inline centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala]: We shouldn't ever need to change `path` to be GCS. That's for the localized path on the VM (vs the universal resource **Location** being a fully qualified, accessible path)

File: centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
+package centaur.cwl
+import io.circe.optics.JsonPath
+import io.circe.optics.JsonPath._
+import io.circe.{Json, JsonObject, yaml}
+
+/**
+  * Tools to pre-process the CWL workflows and inputs before feeding them to Cromwell so they can be executed on PAPI.
+  */
+object PAPIPreprocessor {
```
Comment (kshakir) [inline centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala]: ToL: Awesome. I'll refactor this into a generic processor that can feed in a prefix jic we want to test other backends too.

File: centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  // GCS directory where inputs for conformance tests are stored
+  private val gcsPrefix = "gs://centaur-cwl-conformance/cwl-inputs/"
```
[the remark this answers is not in the document]
Comment (Horneth) [reply centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala]: editors and owners of our (Cromwell dev) google project

File: centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala (hunk 1 above)
Comment (kshakir) [reply centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala]: The above comment is already an open ticket if someone wants to pick it up: #3132

**Comment by Horneth** [comment]
@cjllanwarne I removed the `path` prefixing and added tests for `location` in `default`.
Also added the file prefixing to CWL workflows (was only for input files before). If @danbills's PR on cwltool gets merged and we can salad CWL workflows with gcs path it should unlock at least another test I believe.

File: centaurCwlRunner/src/main/scala/centaur/cwl/CentaurCwlRunner.scala [hunk 4]
```diff
@@ -113,8 +115,8 @@ object CentaurCwlRunner extends StrictLogging {
     }
     val outdirOption = args.outdir.map(_.pathAsString)
     val testName = workflowPath.name
-    val workflowContents = parsedWorkflowPath.contentAsString
-    val inputContents = args.workflowInputs.map(_.contentAsString)
+    val workflowContents = preProcessIfNotLocal(PAPIPreprocessor.preProcessWorkflow)(parsedWorkflowPath.contentAsString)
+    val inputContents = args.workflowInputs.map(_.contentAsString) map preProcessIfNotLocal(PAPIPreprocessor.preProcessInput)
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply centaurCwlRunner/src/main/scala/centaur/cwl/CentaurCwlRunner.scala]: ugh oops nm 😦

**Pull request merged**